proposed/2021/NuGetConsumePDBInPackageReference.md

```diff
@@ -0,0 +1,216 @@
+
+## Goals
+
+* Define an approach to make .NET SDK be able to consume .pdb, .xml files.
+
+## Non-Goals
+
+* .NET SDK side change to consume .pdb or .xml.
+
+## Solution
+
+For any given assembly under lib and runtime folder from a package reference, if there are files next to it that differ only by extension, NuGet will add a "related" property underneath the assembly in targets section of assets file, listing the extensions of these files, seperated by `;`.
+
+* Applies to all the compile time assemblies and runtime assemblies.
+* Apply to transitive dependencies.
+* Only apply to package reference. Will not apply to project reference.
+
+
+### Solution - Technical details
+
+#### Package reference:
+For example, one project has a package reference of PackageA. Suppose in folder ./PackageA/1.0.0/lib/netstandard2.0, there are PackageA.pdb, PackageA.xml file besides PackageA.dll.
+Currently, the targets section in assets file is:
+
+```json
+{
+  "version": 3,
+  "targets": {
+    ".NETCoreApp,Version=v5.0": {
+      "PackageA/1.0.0": {
+        "type": "package",
+        "compile": {
+           "lib/netstandard2.0/PackageA.dll": {}
+        },
+        "runtime": {
+           "lib/netstandard2.0/PackageA.dll": {}
+      }
+    }
+  },
+```
+When this solution is applied, the targets section in assets file will be:
+
+```json
+{
+  "version": 3,
+  "targets": {
+    ".NETCoreApp,Version=v5.0": {
+      "PackageA/1.0.0": {
+        "type": "package",
+        "compile": {
+           "lib/netstandard2.0/PackageA.dll": {
+              "related": "pdb;xml"
+           }
+        },
+        "runtime": {
+           "lib/netstandard2.0/PackageA.dll": {
+              "related": "pdb;xml"
+           }
+      }
+    }
+  },
+```
+#### Project reference:
+For example, one project has a project reference of ProjectA. Supppose in folder ./ProjectA/lib/net5.0, there are PackageA.pdb, PackageA.xml file besides ProjectA.dll.
+There will be no change in targets section in assets file for project reference:
+
+```json
+{
+  "version": 3,
+  "targets": {
+    ".NETCoreApp,Version=v5.0": {
+        "ClassLibrary1/1.0.0": {
+        "type": "project",
+        "framework": ".NETCoreApp,Version=v5.0",
+        "compile": {
+          "bin/placeholder/ClassLibrary1.dll": {}
+        },
+        "runtime": {
+          "bin/placeholder/ClassLibrary1.dll": {}
+        }
+      }
+    },
+```
+
+
+
+On the library side there are no changes. 
+
+This solution is adding a `related` property in [LockFileItem](https://github.com/NuGet/NuGet.Client/blob/4fef99532f4022504feec5f68c8501cbeadd3aed/src/NuGet.Core/NuGet.ProjectModel/LockFile/LockFileItem.cs). The [LockFileItem](https://github.com/NuGet/NuGet.Client/blob/4fef99532f4022504feec5f68c8501cbeadd3aed/src/NuGet.Core/NuGet.ProjectModel/LockFile/LockFileItem.cs) type which represents an element in the compile list already has a collection of properties. 
```

Comment:
At first I didn't understand why the library didn't require changes, but it was explained by the next paragraph. Pulling them together would be clearer I think.

```suggestion
On the library side there are no changes. This solution is adding a `related` property in [LockFileItem](https://github.com/NuGet/NuGet.Client/blob/4fef99532f4022504feec5f68c8501cbeadd3aed/src/NuGet.Core/NuGet.ProjectModel/LockFile/LockFileItem.cs). The [LockFileItem](https://github.com/NuGet/NuGet.Client/blob/4fef99532f4022504feec5f68c8501cbeadd3aed/src/NuGet.Core/NuGet.ProjectModel/LockFile/LockFileItem.cs) type which represents an element in the compile list already has a collection of properties.
```

Reply:
Thanks! Applied.